pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -220,10 +228,43 @@ func sortedKeys(m map[string]string) []string {
 	return keys
 }
 
-// getUserData returns the exact same string for equivalent input,
+func (p *LaunchTemplateProvider) getBottleRocketUserData(ctx context.Context, constraints *v1alpha1.Constraints, additionalLabels map[string]string) (string, error) {
+	var userData string
+	userData += fmt.Sprintf(`[settings.kubernetes]
+cluster-name = "%s"
+api-server = "%s"
+`,
+		injection.GetOptions(ctx).ClusterName,
+		injection.GetOptions(ctx).ClusterEndpoint)
+
+	if constraints.KubeletConfiguration.ClusterDNS != nil {
+		userData += fmt.Sprintf("cluster-dns-ip = \"%s\"", constraints.KubeletConfiguration.ClusterDNS)
+	}
+
+	caBundle, err := p.GetCABundle(ctx)
```

Comment:
optional:

You might be able to collapse a fair amount of complexity by getting the CABundle outside of both userdata generation methods and passing it in. This means these functions will no longer error and no longer need error handling logic.

Reply:
optional:

Looking deeper -- it really shouldnt be possible for GetCABundle to error either, since we're just reading a static file. We could consider panicing on error and collapsing all of the error handling.

Reply:
I left this for another day - Didn't wanna bite off too much at once. :)